src/Platforms/OMPTarget/CMakeLists.txt

```diff
@@ -22,5 +22,35 @@ ELSE()
 ENDIF()
 
 IF(NOT QMC_COMPLEX)
-  TARGET_COMPILE_DEFINITIONS(platform_omptarget PRIVATE -DOPENMP_NO_COMPLEX)
+  TARGET_COMPILE_DEFINITIONS(platform_omptarget PUBLIC OPENMP_NO_COMPLEX)
+else()
+  # Test if C++ compiler supports OpenMP user defined reduction on complex type
+  try_compile(OMP_UDR_COMPLEX_OKAY ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/test_user_defined_reduction_complex.cpp
+              CMAKE_FLAGS "${CMAKE_CXX_FLAGS}"
+              OUTPUT_VARIABLE COMPILE_OUTPUT)
+
+  if (NOT OMP_UDR_COMPLEX_OKAY)
+    set(COMPILE_FAIL_OUTPUT omp_UDR_complex_fail.txt)
+    file(WRITE "${CMAKE_BINARY_DIR}/${COMPILE_FAIL_OUTPUT}" "${COMPILE_OUTPUT}")
+    message(STATUS "OpenMP user-defined reduction functionality compilation check failed!"
```

Comment:
WARNING and not STATUS?

Reply:
Because the next step makes an attempt to try the implicit way. So it is a recoverable failure and always print when using NVHPC. So I prefer not to print a scary warning.